app/javascript/components/challenges/Login.tsx

```diff
@@ -1,7 +1,28 @@
 import * as React from 'react';
+import { useState } from 'react';
+
+
 
 export default function Login() {
-  const [email, setEmail] = React.useState<string>('');
+
+  const [email, setEmail] = useState("");
+  const [emailValidationError, setEmailValidationError] = useState("");
+  const [showPassword, setShowPassword] = useState<boolean>(true);
+
+
+  function handleEmailChange(event) {
+    setEmail(event.target.value);
+  };
+
+  function validateEmail(value) {
+    let error;
+    if (!value) {
+      error = 'Required';
+    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value)) {
+      error = 'Invalid email address';
+    }
+    return error;
+  };
```

Comment:
Pretty much there with this one.

Add an onblur handler to the email input and add the email validation error into the error box:

```
<input name="email" type="email" value={email} onChange={handleEmailChange} onBlur={validateEmail} />
<div style={{ color: 'red', margin: '10px 0' }}>{emailValidationError}</div>
```

Then change your validation function slightly:

```
function validateEmail(event) {
    const value = event.target.value;
    let error;
    if (!value) {
      error = 'Required';
    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,6}$/i.test(value)) {
      error = 'Invalid email address';
    }
    setEmailValidationError(error)
  };
```